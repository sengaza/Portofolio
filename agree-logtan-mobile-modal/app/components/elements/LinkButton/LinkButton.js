import React, { Component } from 'react';
import { Text } from 'react-native';

import styles from './LinkButton.styles';
import { noop } from '../../../utils';

class LinkButton extends Component {
  render() {
    const { title, onPress = noop, customButtonStyle } = this.props;
    const buttonStyle = styles.button;

    return (
      <Text onPress={onPress} style={[buttonStyle, customButtonStyle]}>
        {title}
      </Text>
    );
  }
}

export default LinkButton;
