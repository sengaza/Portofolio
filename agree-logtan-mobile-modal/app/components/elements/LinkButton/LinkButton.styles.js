import { StyleSheet } from 'react-native';

import { COLOR_BASE_PRIMARY_MAIN } from '../../../styles';

const button = {
  height: 40,
  width: 112,
  textAlign: 'center',
  textAlignVertical: 'center',
  color: COLOR_BASE_PRIMARY_MAIN
};

export default StyleSheet.create({
  button
});
