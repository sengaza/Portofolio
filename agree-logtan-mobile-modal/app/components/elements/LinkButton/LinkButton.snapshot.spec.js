import React from 'react';
import renderer from 'react-test-renderer';

import LinkButton from './LinkButton';

jest.mock('./LinkButton.styles.js');
jest.mock('Text', () => 'Text');

test('LinkButton renders correctly', () => {
  const tree = renderer.create(<LinkButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('LinkButton renders correctly when text is set', () => {
  const tree = renderer.create(<LinkButton title="Hello" />).toJSON();
  expect(tree).toMatchSnapshot();
});
