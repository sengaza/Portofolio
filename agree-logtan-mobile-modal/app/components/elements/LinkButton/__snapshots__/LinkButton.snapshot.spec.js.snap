// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders correctly 1`] = `
<Text
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#1ea54f",
        "height": 40,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 112,
      },
      undefined,
    ]
  }
/>
`;

exports[`LinkButton renders correctly when text is set 1`] = `
<Text
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": "#1ea54f",
        "height": 40,
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": 112,
      },
      undefined,
    ]
  }
>
  Hello
</Text>
`;
